ui/src/app/cohort-search/attributes-page-v2/attributes-page-v2.component.tsx

```diff
@@ -0,0 +1,609 @@
+        resp.items.forEach(attr => {
+          if (attr.type === AttrName[AttrName.NUM]) {
+            // NUM attributes set the min and max range for the number inputs in the attributes form
+            if (!form.num.length) {
+              form.num.push({
+                name: AttrName.NUM,
+                operator: this.isSurvey ? 'ANY' : null,
+                operands: [],
+                conceptId: conceptId,
+                [attr.conceptName]: parseInt(attr.estCount, 10)
+              });
+            } else {
+              form.num[0][attr.conceptName] = parseInt(attr.estCount, 10);
+            }
+          } else {
+            // CAT attributes are displayed as checkboxes in the attributes form
+            if (parseInt(attr.estCount, 10) > 0) {
+              attr['checked'] = false;
+              form.cat.push(attr);
+            }
+          }
+        });
+        const count = this.isSurvey ? this.nodeCount : null;
+        this.setState({count, form, loading: false});
+      });
+    }
+
+    toggleCheckbox(checked: boolean) {
+      const {form} = this.state;
+      let {node: {count}} = this.props;
+      if (checked) {
+        form.exists = true;
+        form.num = form.num.map(attr =>
+          ({...attr, operator: this.isPhysicalMeasurement ? 'ANY' : null, operands: []}));
+        form.cat = form.cat.map(attr => ({...attr, checked: false}));
+      } else {
+        count = null;
+        form.exists = false;
+      }
+      this.setState({form, count});
+    }
+
+    selectChange(attributeIndex: number, value: string) {
+      const {form} = this.state;
+      form.num[attributeIndex].operator = value;
+      if (this.isBloodPressure) {
+        // for blood pressure, either both operators have to be 'ANY' OR neither can be 'ANY'
+        const otherAttribute = attributeIndex === 0 ? 1 : 0;
+        if (value === 'ANY') {
+          form.num[otherAttribute].operator = 'ANY';
+          form.num[otherAttribute].operands = form.num[attributeIndex].operands = [];
+        } else if (form.num[otherAttribute].operator === 'ANY') {
+          form.num[otherAttribute].operator = value;
+        }
+      } else if (value === 'ANY') {
+        form.num[attributeIndex].operands = [];
+      }
+      if (value !== Operator[Operator.BETWEEN]) {
+        // delete second operand if it exists
+        form.num[attributeIndex].operands.splice(1);
+      }
+      const count = value === 'ANY' ? this.nodeCount : null;
+      this.setState({form, count});
+    }
+
+    inputChange(input: string, attributeIndex: number, operandIndex: number) {
+      const {form} = this.state;
+      form.num[attributeIndex].operands[operandIndex] = sanitizeNumericalInput(input);
+      this.setState({form, count: null});
+    }
+
+    checkboxChange(checked: boolean, index: number) {
+      const {form} = this.state;
+      form.cat[index].checked = checked;
+      this.setState({form});
```

Comment:
Optional: Curious if we can use loadash here like: 
eg. var object = { 'a': [{ 'b': { 'c': 3 } }] }; 
_.set(object, 'a[0].b.c', 4);

Reply:
Seems like it should work but I couldn't get it to actually update the form object. Tried like this:
```
fp.set(`cat[${index}].checked`, checked, form);
```
([fp syntax](https://gist.github.com/jfmengels/6b973b69c491375117dc#_setpath-value-object) switches up the args: `_.set(path, value, object)`)